src/main/java/org/springframework/social/salesforce/api/SalesforceUserDetails.java

```diff
@@ -0,0 +1,50 @@
+package org.springframework.social.salesforce.api;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import java.io.Serializable;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SalesforceUserDetails implements Serializable {
```

Comment:
Looking at the the response from both of the endpoints, I think it might make more sense to make SalesforceUserDetails an extension of SalesforceProfile - I think SalesforceProfile would be better renamed though, as it contains User Detail, not User Profile. I added two classes initially because the entities returned by each endpoint seem to be conceptually different, and I though it might be easier to deserialize them this way - but for the purpose of this change I think the latest implementation would be enough. Most of the details are not used anyway.
Let me know if the current implementations makes sense, otherwise changing SalesforceProfile to be an extension of SalesforceUserDetails would not be complicated. Thanks.